lib/services/unified_resource.go

```diff
@@ -127,7 +127,7 @@ func (cfg *UnifiedResourceCacheConfig) CheckAndSetDefaults() error {
 	return nil
 }
 
-func (c *UnifiedResourceCache) putLocked(resource resource) {
+func (c *UnifiedResourceCache) putLocked(resource CachedUnifiedResource) {
```

Comment:
how does this work when there are multiple database servers for the same database?

Reply:
> how does this work

how does _what_ work?

Reply:
how does **`UnifiedResourceCache`** work when handling resource types that can have multiple heartbeats from HA setup.

Reply:
ah, i see. the UnifiedResourceCache was originally made to support the resources supported by the web UI. for this example, ill talk about databases.

the web UI only cares about the database, but since those don't heartbeat we have to watch DatabaseServers. essentially, when we receive a heartbeat for some "container" type, we store it by creating the resource key based on the "contained" type. so if there are multiple DatabaseServers, we basically just store the "database" as the latest DatabaseServer that heartbeat. that way when the web UI fetches databases, its really getting a "deduped" list of database servers by unique contained database

Reply:
to answer you @GavinFrazar , what I'd like to do, is compliment the above by also storing _all_ of the database servers in someway that can be fetched by the "contained" database. so if i have 3 db servers serving `database1234`, it'd be cool if i could have something like `database1234->[dbServer1, dbServer2, dbServer3]` which would make this fetching easier

Reply:
got it thanks! 
To handler health check properly in this case, we either have to store a list of database_server heartbeats or somehow combine the health statuses, as we need to represent health status from all servers in the WebUI